### PR TITLE
Add step by step example of how to use scaladoc

### PR DESCRIPTION
**Pull request by kpodsiad**

### CODE REVIEW

**Comment by kpodsiad** [comment]
@pikinier20 IMO it would be nice to add an example of how to host docs using e.g. Github pages. Maybe there should be some repository where everything is configured and the user can see how it works & looks. What do you think?

File: docs/docs/usage/scaladoc/summary-example.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+## View documentation locally
+Generated docs are store at `target/scala-<version>` directory. Let's change target directory to the more toplevel one.
+
+```scala
+lazy val docSettings: Seq[Setting[_]] = Seq(
+  Compile / doc / scalacOptions ++= Seq(
+    Compile / doc / target := file("generated-docs"),
+    "-project", "scaladoc example",
+  )
```
Comment (BarkingBad) [inline docs/docs/usage/scaladoc/summary-example.md:42]: Wouldn't it be more convenient to not nest the settings

```suggestion
  Compile / doc / target := file("generated-docs"),
  Compile / doc / scalacOptions ++= Seq(
    "-project", "scaladoc example",
  )
```

File: docs/docs/usage/scaladoc/summary-example.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+)
+```
+
+However, so generated documentation is missing a single entry point.
```
Comment (BarkingBad) [inline docs/docs/usage/scaladoc/summary-example.md:46]: Hopefully, this should be fixed in `3.1.0`. Can you add some tip note here?

**Review by BarkingBad — COMMENTED** [review]
Looks OK

File: docs/docs/usage/scaladoc/summary-example.md [hunk 3]
```diff
@@ -36,14 +36,14 @@ Generated docs are store at `target/scala-<version>` directory. Let's change tar
...
+  Compile / doc / target := file("generated-docs"),
   Compile / doc / scalacOptions ++= Seq(
-    Compile / doc / target := file("generated-docs"),
     "-project", "scaladoc example",
   )
 )
 ```
 
-However, so generated documentation is missing a single entry point. 
+However, in scaladoc `3.2.0` and earlier, so generated documentation is missing a single entry point. This is no longer case in `3.1.x` and hence `"-siteroot", "docs"` is not needed anymore.
```
Comment (BarkingBad) [inline docs/docs/usage/scaladoc/summary-example.md:46]:
```suggestion
However, in scaladoc `3.0.2` so generated documentation is missing a single entry point. This is no longer case in `3.1.x` and hence `"-siteroot", "docs"` is not needed anymore.
```

It was a regression brought by `-Yapi-subdirectory` flag introduced in `3.0.2`

**Review by BarkingBad — CHANGES_REQUESTED** [review]

**Comment by ckipp01** [comment]
Since there hasn't been any activity here in quite some time and much of this is contained in https://docs.scala-lang.org/scala3/guides/scaladoc/index.html, I'll go ahead and close this. If you feel like anything is still missing from the docs, feel free to port them.

**Pull request closed without merging**